python-lib/dku_stress_test_center/stress_test_center.py

```diff
@@ -0,0 +1,193 @@
+
+            pertubed_df.loc[:, target_column] = yt
+
+            pertubed_df[DkuStressTestCenterConstants.STRESS_TEST_TYPE] = stress_test_id
+            pertubed_df[DkuStressTestCenterConstants.DKU_ROW_ID] = pertubed_df.index
+
+            # probably need to store the shifted_indices
+
+            perturbed_datasets_df = perturbed_datasets_df.append(pertubed_df)
+
+        return perturbed_datasets_df
+
+
+def build_stress_metric(y_true: np.array,
+                        y_pred: np.array,
+                        stress_test_indicator: np.array,
+                        pos_label: str or int):
+    # batch evaluation
+    # return average accuracy drop
+    # return average f1 drop
+    # return robustness metrics: 1-ASR or imbalanced accuracy for prior shift
+
+    metrics_df = pd.DataFrame(columns=[DkuStressTestCenterConstants.STRESS_TEST_TYPE,
+                                       DkuStressTestCenterConstants.ACCURACY_DROP,
+                                       DkuStressTestCenterConstants.F1_DROP,
+                                       DkuStressTestCenterConstants.ROBUSTNESS])
+
+    clean_filter = stress_test_indicator == DkuStressTestCenterConstants.CLEAN
+    clean_y_true = y_true[clean_filter]
+    clean_y_pred = y_pred[clean_filter]
+    clean_accuracy = accuracy_score(clean_y_true, clean_y_pred)
+    clean_f1_score = f1_score(clean_y_true, clean_y_pred, pos_label=pos_label)
+
+    stress_ids = np.unique(stress_test_indicator)
+    stress_ids = np.delete(stress_ids, np.where(stress_ids==DkuStressTestCenterConstants.CLEAN))
+    for stress_id in stress_ids:
+        stress_filter = stress_test_indicator == stress_id
+        stress_y_true = y_true[stress_filter]
+        stress_y_pred = y_pred[stress_filter]
+
+        stress_accuracy = accuracy_score(stress_y_true, stress_y_pred)
+        stress_f1_score = f1_score(stress_y_true, stress_y_pred, pos_label=pos_label)
+
+        stress_acc_drop = stress_accuracy - clean_accuracy
+        stress_f1_drop = stress_f1_score - clean_f1_score
+
+        if stress_id == DkuStressTestCenterConstants.PRIOR_SHIFT:
+            robustness = balanced_accuracy_score(stress_y_true, stress_y_pred)
+        else:
+            # 1 - Attack Success Rate
+            clean_correctly_predicted_filter = clean_y_pred == clean_y_true
+            robustness = accuracy_score(clean_y_pred[clean_correctly_predicted_filter],
+                                        stress_y_pred[clean_correctly_predicted_filter])
+
+        metrics_df = metrics_df.append({
+            DkuStressTestCenterConstants.STRESS_TEST_TYPE: stress_id,
+            DkuStressTestCenterConstants.ACCURACY_DROP: stress_acc_drop,
+            DkuStressTestCenterConstants.F1_DROP: stress_f1_drop,
+            DkuStressTestCenterConstants.ROBUSTNESS: robustness
+        }, ignore_index=True)
+
+    return metrics_df
+
+
+def get_critical_samples(y_true_class_confidence: np.array,  # n_rows x 1
```

Comment:
This function should return a df with *two* column: `_dku_row_identifier_` and `uncertainty` 
Right now it returns a df with *one* column `uncertainty` and `_dku_row_identifier_ ` as index, which is not ideal